centreon/src/Core/TimePeriod/Domain/Model/TimeRange.php

```diff
@@ -42,6 +47,13 @@ class TimeRange implements \Stringable
      */
     public function __construct(string $timeRange)
     {
+
+        $resolvedTimeRangeAlias = $this->resolveTimeRangeAlias($timeRange);
+        if ($resolvedTimeRangeAlias) {
+            return;
+        }
+
+        $timeRange = $this->timeRange;
         Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');
 
         if (! $this->isValidTimeRangeFormat($timeRange)) {
```

Comment:
Seen together : 
* delete your method resolveTimeRangeAlias()
* move in the method isValidTimeRangeFormat(), the following code :  `Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');` at the begining
* in the method isValidTimeRangeFormat(), before :  `Assertion::minLength($timeRange, 11, 'TimeRange::timeRange');` write this code :
```
if (empty($timeRange)) {
    return true;
}
```
* delete your changes in the unit test
* enjoy ! :D

Reply:
resolved

Reply:
good